var/spack/repos/builtin/packages/igraph/package.py

```diff
@@ -6,12 +6,49 @@
 from spack.package import *
 
 
-class Igraph(AutotoolsPackage):
+class Igraph(CMakePackage, AutotoolsPackage):
     """igraph is a library for creating and manipulating graphs."""
 
     homepage = "https://igraph.org/"
     url = "https://github.com/igraph/igraph/releases/download/0.7.1/igraph-0.7.1.tar.gz"
 
+    version("0.10.6", sha256="99bf91ee90febeeb9a201f3e0c1d323c09214f0b5f37a4290dc3b63f52839d6d")
     version("0.7.1", sha256="d978030e27369bf698f3816ab70aa9141e9baf81c56cc4f55efbe5489b46b0df")
 
+    variant("shared", default=False, description="Enable shared build")
+
+    build_system(
+        conditional("cmake", when="@0.9:"),
+        conditional("autotools", when="@:0.9"),
```

Comment:
```suggestion
        conditional("autotools", when="@:0.8"),
```
Version ranges in Spack are inclusive. So 0.8 is the most recent version that supports Autotools